src/EditorFeatures/Text/Shared/Extensions/ITextSnapshotLineExtensions.cs

```diff
@@ -43,11 +35,10 @@ internal static class ITextSnapshotLineExtensions
         {
             Contract.ThrowIfNull(line);
 
-            var text = line.GetText();
-
-            for (var i = 0; i < text.Length; i++)
+            var snapshot = line.Snapshot;
+            for (var i = 0; i < line.Length; i++)
```

Comment:
>for (var i = 0; i < line.Length; i++) [](http://example.com/codeflow?start=12&length=37)

nit: might be a bit simpler to iterate over the line extents. eg:

```
var snapshot = line.Snapshot;

for (var i = line.Start.Position; i < line.End.Position; i++)
{
    if (!char.IsWhiteSpace(snapshot[i]))
    {
        return i;
    }
}
```

Reply:
good idea, fixed

Reply:
@olegtk @ToddGrun I think this broke the code -- it's now returning the position in the snapshot, not the offset from the start of the line.

Reply:
yeah, tests are failing, fixing it now

Reply:
sorry @ToddGrun , I reverted the change, I think this way is OK too

Reply:
Oh yeah, it should have been switched to return "i - line.Start.Position" as it returns an offset. The current code is probably better than the modified version of what I proposed then, as I didn't notice it was an offset method.